brave-okhttp/src/test/java/com/github/kristofa/brave/okhttp/BraveTracingInterceptorTest.java

```diff
@@ -0,0 +1,285 @@
+
+    assertThat(collectedSpans()).containsExactly(
+        span.toBuilder()
+            .addBinaryAnnotation(BinaryAnnotation.create(TraceKeys.HTTP_STATUS_CODE, "404", local))
+            .build()
+    );
+  }
+
+  @Test
+  public void reportsToZipkin_IncludesQueryParams() throws Exception {
+    server.enqueue(new MockResponse());
+
+    HttpUrl url = server.url("foo?z=2&yAA");
+    client.newCall(new Request.Builder().url(url).build()).execute();
+
+    assertThat(collectedSpans()).containsExactly(
+        span.toBuilder().binaryAnnotations(asList(
+            BinaryAnnotation.create(TraceKeys.HTTP_URL, url.toString(), local))
+        ).build()
+    );
+  }
+
+  @Test
+  public void reportsToZipkin_requestTagIgnoredByDefault() throws Exception {
+    server.enqueue(new MockResponse());
+
+    HttpUrl url = server.url("foo");
+    client.newCall(new Request.Builder().url(url).tag("foo").build()).execute();
+
+    assertThat(collectedSpans()).containsExactly(span);
+  }
+
+  @Test
+  public void preferTagAsSpanName() throws Exception {
+    close();
+    interceptor = interceptorBuilder(Sampler.ALWAYS_SAMPLE).preferTagAsSpanName(true).build();
+    client = new OkHttpClient.Builder().addInterceptor(interceptor).build();
+
+    server.enqueue(new MockResponse());
+
+    HttpUrl url = server.url("foo");
+    client.newCall(new Request.Builder().url(url).tag("foo").build()).execute();
+
+    assertThat(collectedSpans())
+        .containsExactly(span.toBuilder().name("foo").build());
+  }
+
+  @Test
+  public void preferTagAsSpanName_noTag() throws Exception {
+    close();
+    interceptor = interceptorBuilder(Sampler.ALWAYS_SAMPLE).preferTagAsSpanName(true).build();
+    client = new OkHttpClient.Builder().addInterceptor(interceptor).build();
+
+    server.enqueue(new MockResponse());
+
+    HttpUrl url = server.url("foo");
+    client.newCall(new Request.Builder().url(url).build()).execute();
+
+    assertThat(collectedSpans()).containsExactly(span);
+  }
+
+  @Test
+  public void whenNetworkInterceptor_preferTagAsSpanName_noTag() throws Exception {
+    close();
+    interceptor = interceptorBuilder(Sampler.ALWAYS_SAMPLE).preferTagAsSpanName(true).build();
+    client = new OkHttpClient.Builder().addNetworkInterceptor(interceptor).build();
+
+    server.enqueue(new MockResponse());
+
+    HttpUrl url = server.url("foo");
+    client.newCall(new Request.Builder().url(url).build()).execute();
+
+    assertThat(collectedSpans()).containsExactly(span);
+  }
+
+  @Test
+  public void whenApplicationInterceptor_followupsAreIgnored() throws Exception {
+    server.enqueue(new MockResponse().setResponseCode(408));
+    server.enqueue(new MockResponse());
+
+    HttpUrl url = server.url("foo");
+    client.newCall(new Request.Builder().url(url).build()).execute();
+
+    assertThat(collectedSpans()).containsExactly(span); // 408 one is ignored
+  }
+
+  @Test
+  public void whenNetworkInterceptor_followupsAreNewSpans() throws Exception {
```

Comment:
new test for when this is a network interceptor. each follow-up (such as a redirect) will show up in zipkin